src/Payments/Integrations/Stripe.php

```diff
@@ -104,12 +105,22 @@ public function get_payment_method_data() {
 	 * @return bool True if merchant allows shopper to save card (payment method) during checkout).
 	 */
 	private function get_allow_saved_cards() {
-		$saved_cards = isset( $this->settings['saved_cards'] ) ? $this->settings['saved_cards'] : false;
+		return isset( $this->settings['saved_cards'] ) ? 'yes' === $this->settings['saved_cards'] : false;
+	}
+
+	/**
+	 * Determine if the payment method requires saving. If saved_cards is set to yes and filters hide the save
+	 * payment method checkbox, we can assume that this method requires saving.
+	 *
+	 * @return bool True if the save payment checkbox should be displayed to the user.
+	 */
+	private function get_requires_saving() {
+		$saved_cards = $this->get_allow_saved_cards();
 		// This assumes that Stripe supports `tokenization` - currently this is true, based on
 		// https://github.com/woocommerce/woocommerce-gateway-stripe/blob/master/includes/class-wc-gateway-stripe.php#L95 .
 		// See https://github.com/woocommerce/woocommerce-gateway-stripe/blob/ad19168b63df86176cbe35c3e95203a245687640/includes/class-wc-gateway-stripe.php#L271 and
 		// https://github.com/woocommerce/woocommerce/wiki/Payment-Token-API .
-		return apply_filters( 'wc_stripe_display_save_payment_method_checkbox', filter_var( $saved_cards, FILTER_VALIDATE_BOOLEAN ) );
+		return ! apply_filters( 'wc_stripe_display_save_payment_method_checkbox', filter_var( $saved_cards, FILTER_VALIDATE_BOOLEAN ) );
```

Comment:
hmm this seems to indicate to me that I might be confused about what `requiresSaving` means. I'm a bit wary of the implicit connection between `requiresSaving` and `allowsSavedCards`. So essentially what we're saying here is that if anything filters this option so it returns false, that means the the saved payment methods ARE required (double negative resulting in that).  I wonder if we should introduce a new filter that allows for explicitly setting this rather than implicitly?

Reply:
> I wonder if we should introduce a new filter that allows for explicitly setting this rather than implicitly?

double down on this.

Reply:
Thomas and I chatted about this in our call and reviewed existing behaviour in the shortcode flow. It turns out we think we can improve this by retaining the explicit use of the filtered value and non filtered value and update the name of the properties to make it more clear how things are used. He's in the process of doing that now.